### PR TITLE
Closes (#121)

### PR DESCRIPTION
**Pull request by sandstone991**
## 🛠️ Fixes Issue
Debounce Isochrone API calls  #121 

## 👨‍💻 Changes proposed

Changed the UNSAFE_ComponentWillUpdate lifecycle which was ran on every change to see if the sliders values had changed to send a new request, made the component call a debounced fetch function on change instead.
As for the sliders they make request only on the mouseUp event (user let go of the slider).

## 📄 Note to reviewers

I encounted two bugs (unrelated to the task) while working on this component.

- The max value of the range is set to 120 
image
which gives error code 151 when calling Valhalla's API (Exceeded max time 100) so I'm guessing the max value has to change to 100 here

- Some requests return 504 which shows the user this messege 
image
which is unhelpful IMO

I'm also dissatisfied with how async calls are done within this component, given unfortuante enough cirumstances a race condition like this can happen 
image

but I think that have to wait until the components are rewritten into functional components

Also please consider this a pre-gsoc pr

### CODE REVIEW

**Comment by sandstone991** [comment]
@nilsnolde kindly review this

File: src/Controls/Isochrones/Waypoints/index.jsx [hunk 1]
```diff
@@ -233,7 +237,7 @@ class Waypoints extends Component {
                 loading={isFetching}
                 results={geocodeResults}
                 value={userInput}
-                onKeyPress={(event: React.KeyboardEvent) => {
```
Comment (chrstnbwnkl) [inline src/Controls/Isochrones/Waypoints/index.jsx:236]: thanks for catching this, shouldn't have made it into master in the first place

File: src/Controls/Isochrones/Waypoints/index.jsx [hunk 2]
```diff
@@ -85,18 +88,18 @@ class Waypoints extends Component {
...
-    const nextMaxRange = nextProps.isochrones.maxRange
-    const nextInterval = nextProps.isochrones.interval
-    if (
-      !R.equals(maxRange, nextMaxRange) ||
-      !R.equals(interval, nextInterval)
-    ) {
-      dispatch(makeIsochronesRequest())
-    }
-  }
+  // componentDidUpdate(nextProps) {
```
Comment (chrstnbwnkl) [inline src/Controls/Isochrones/Waypoints/index.jsx:91]: don't just uncomment unused code, remove it instead

File: src/Controls/Isochrones/Waypoints/index.jsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 import { connect } from 'react-redux'
-import * as R from 'ramda'
+// import * as R from 'ramda'
```
Comment (chrstnbwnkl) [inline src/Controls/Isochrones/Waypoints/index.jsx:4]: should be removed as well

**Review by chrstnbwnkl — CHANGES_REQUESTED** [review]
Just some minor comments

File: src/Controls/Isochrones/Waypoints/index.jsx (hunk 2 above)
Comment (sandstone991) [reply src/Controls/Isochrones/Waypoints/index.jsx:91]: sure, no problem

**Pull request merged**